File: backend/ConstructionManagement/Entities/Attachment.cs
namespace ConstructionManagement.Entities
{
    public class Attachment
    {
        public Guid AttachmentId { get; set; }

        public Guid ProjectId { get; set; }

        public required string FileName { get; set; }

        public required string FilePath { get; set; }

        public string? ContentType { get; set; }

        public long FileSize { get; set; }

        public DateTime UploadedAt { get; set; } = DateTime.UtcNow;

        public Project Project { get; set; } = null!;
    }
}

File: frontend/src/types/attachment.ts
export interface Attachment {
  attachmentId: string
  projectId: string
  fileName: string
  filePath: string
  contentType?: string | null
  fileSize: number
  uploadedAt: string
}

export interface CreateAttachmentPayload {
  fileName: string
  filePath: string
  contentType?: string | null
}

export interface UpdateAttachmentPayload {
  fileName: string
  filePath: string
  contentType?: string | null
}
